controllers/message.go

```diff
@@ -224,7 +232,7 @@ func (c *ApiController) AddMessage() {
 		return
 	}
 
-	if success {
+	if success && addMessageAfterSuccess {
```

Comment:
Why not return early?

Reply:
We need to execute object.AddMessage() at line 240. However, object.AddMessage() at line 267 shouldn't be executed.  We could just RETURN after line 240, but it would require "if" block one more time, and I guess that's not as clean as this.